### PR TITLE
refactored hubpy

### PR DESCRIPTION
**Pull request by ManushiM**
added new submodules for each object
basic cleanup to remove old references.

### CODE REVIEW

**Comment by ajturner** [comment]
@ManushiM it looks like Hub.py still has all the old code. That's now duplicated with the extracted methods. 

Please update Hub.py to remove the duplication.

**Comment by ManushiM** [comment]
Thanks, I made changes to hub.py and enhanced a few user guides too.

File: examples/For arcgishub library/.ipynb_checkpoints/Searching and Filtering content from your site-checkpoint.ipynb [hunk 1]
```diff
@@ -0,0 +1,383 @@
...
+   "metadata": {},
+   "outputs": [
+    {
+     "ename": "AttributeError",
+     "evalue": "'NoneType' object has no attribute 'typeKeywords'",
+     "output_type": "error",
+     "traceback": [
+      "\u001b[1;31m---------------------------------------------------------------------------\u001b[0m",
+      "\u001b[1;31mAttributeError\u001b[0m                            Traceback (most recent call last)",
+      "\u001b[1;32m~\\AppData\\Local\\Temp\\1\\ipykernel_23116\\3436182706.py\u001b[0m in \u001b[0;36m<module>\u001b[1;34m\u001b[0m\n\u001b[1;32m----> 1\u001b[1;33m \u001b[0mdc\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mmyhub\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0msites\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mget\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;34m'3adff78f286e4ecf936bdefdfeca9890'\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m      2\u001b[0m \u001b[0mdc\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
```
Comment (ajturner) [inline examples/For arcgishub library/.ipynb_checkpoints/Searching and Filtering content from your site-checkpoint.ipynb:51]: why is this error here in the notebook?

it looks like some of the rest of the script didn't work.

**Review by ajturner — CHANGES_REQUESTED** [review]

**Comment by ManushiM** [comment]
Hmm, odd because I replaced the file last night to one that doesn't throw an error.
I pushed the correct file now and verified that it's error free.

**Comment by ManushiM** [comment]
Another change requires replacing the `requests` library by the in-house `gis._con` object to connect to hub's v3 domain api service. However, that change can only be successfully made and verified after v2.1.0 of the Python API is live. `requests` works perfectly for now and doesn't break any site functionality, but I will make that change in a month or two for increased security while hitting the api endpoint, and for uniformity across the 2 versions of hub-py. It is not a blocker (functionality wise) to merge this PR.

**Comment by brollison** [comment]
@ajturner / @ManushiM where do we stand on getting this large-scale change reviewed and merged (if approved)? Two fixes for escalation requests were [committed](https://github.com/Esri/hub-py/pull/137/commits/10e6a8cd3848a4212977306d722b87b9d14b5876) to this branch.

**Comment by ajturner** [comment]
thanks for ping @brollison . I will look over this week. 

related @ManushiM what is the appropriate way to manage + integrate changes into the Geosaurus builds?

**Comment by ManushiM** [comment]
@ajturner for several months now I have been updating hub in the geosaurus repo first, and the version of hub there is more uptodate.

I'm not aware of any direct way to sync these two repos so that changes in one reflect in another. If changes are made on hub-py that are not already present in geosaurus, a separate PR to geosaurus will need to be made before the code-freeze date of the next release.

**Comment by ajturner** [comment]
This is now complete. 

- fixed to work with multiple Hub environments
- removed Discussions `qa` only limitation
- added exception check for groups

there aren't automated tests (yet) so while I've run through several notebooks it hasn't been exhaustive. 
I plan to task BE devs in 2023Q3 + Q4 with adding tests and maintaining the Hub.py module as part of their API development + delivery.

**Review by ajturner — APPROVED** [review]

**Pull request merged**